packages/interactive_media_ads/test/ios/ads_manager_delegate_tests.dart

```diff
@@ -0,0 +1,267 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:interactive_media_ads/src/ios/interactive_media_ads.g.dart'
+    as ima;
+import 'package:interactive_media_ads/src/ios/interactive_media_ads_proxy.dart';
+import 'package:interactive_media_ads/src/ios/ios_ads_manager_delegate.dart';
+import 'package:interactive_media_ads/src/platform_interface/platform_interface.dart';
+import 'package:mockito/annotations.dart';
+
+@GenerateNiceMocks(<MockSpec<Object>>[MockSpec<ima.IMAAdsManager>()])
+void main() {
+  group('IOSAdsManagerDelegate', () {
+    test('didReceiveAdEvent calls onAdEvent', () {
+      final ima.PigeonInstanceManager instanceManager =
+          ima.PigeonInstanceManager(onWeakReferenceRemoved: (_) {});
+
+      late final void Function(
+        ima.IMAAdsManagerDelegate,
+        ima.IMAAdsManager,
+        ima.IMAAdEvent,
+      ) didReceiveAdEventCallback;
+
+      late final ima.IMAAdsManagerDelegate delegate;
+      final InteractiveMediaAdsProxy imaProxy = InteractiveMediaAdsProxy(
+        newIMAAdsManagerDelegate: ({
+          required void Function(
+            ima.IMAAdsManagerDelegate,
+            ima.IMAAdsManager,
+            ima.IMAAdEvent,
+          ) didReceiveAdEvent,
+          required void Function(
+            ima.IMAAdsManagerDelegate,
+            ima.IMAAdsManager,
+            ima.IMAAdError,
+          ) didReceiveAdError,
+          required void Function(ima.IMAAdsManagerDelegate, ima.IMAAdsManager)
+              didRequestContentPause,
+          required void Function(ima.IMAAdsManagerDelegate, ima.IMAAdsManager)
+              didRequestContentResume,
+        }) {
+          didReceiveAdEventCallback = didReceiveAdEvent;
+          delegate = ima.IMAAdsManagerDelegate.pigeon_detached(
+            didReceiveAdEvent: didReceiveAdEvent,
+            didReceiveAdError: didReceiveAdError,
+            didRequestContentPause: didRequestContentPause,
+            didRequestContentResume: didRequestContentResume,
+            pigeon_instanceManager: instanceManager,
+          );
+          return delegate;
+        },
+      );
+
+      final IOSAdsManagerDelegate adsManagerDelegate = IOSAdsManagerDelegate(
+        IOSAdsManagerDelegateCreationParams(
+          onAdEvent: expectAsync1((AdEvent event) {
+            expect(event.type, AdEventType.allAdsCompleted);
```

Comment:
It looks like these tests only expect within the callback, in which case failing to call the callback wouldn't fail the test. If that's the case, these should all have a completer declared up front, and completed with the expectation, so the end of the test can also assert that the completer completes (and thus the test did call the callback).

Reply:
[expectAsync1](https://api.flutter.dev/flutter/package-matcher_expect/expectAsync1.html) waits for the callback to run before it considers the current test to be complete.

I started using for testing with callback methods with more than 1 parameter since completers can only take 1.

Reply:
Oh, very cool, I didn't even notice that `expect*`. TIL, thanks!